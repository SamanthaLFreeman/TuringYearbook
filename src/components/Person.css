.personContainer {
  background-color: white;
  width: 25%;
}

.image {
  width: 200px;
  height: 200px;
}
