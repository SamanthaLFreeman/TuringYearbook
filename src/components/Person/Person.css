.personContainer {
  background-color: rgba(255, 255, 255, 0.95);
  border-radius: 30px;
  color: #2c2c2c;
  height: 350px;
  margin: 20px 0;
  padding: 40px;
  width: 230px;
}

.image {
  width: 200px;
  height: 200px;
}

.quote {
  font-style: italic;
}

.remove-button {
  width: 200px;
}
