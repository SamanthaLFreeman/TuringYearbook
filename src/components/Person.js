import React from 'react';
import './Person.css';

const Person = ({individual, removeStudent, studentCheck}) => {
  const button = studentCheck === 'student' ? <button onClick={() => removeStudent(individual.id)}>Remove</button> : '';
  const editableCheck = studentCheck === 'student' ? true : false;
  
  return (
    <section className="personContainer" key={individual.id}>
      <img className="image" src={individual.photo} alt={individual.name} />
      <h3 className="name" contentEditable={editableCheck}>{individual.name}</h3>
      <p className="quote" contentEditable={editableCheck}>{individual.quote}</p>
      <p className="superlative" contentEditable={editableCheck}>{individual.superlative}</p>
      {button}
    </section>
  )
}

export default Person;
