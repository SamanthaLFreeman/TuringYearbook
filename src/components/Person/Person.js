import React from 'react';
import './Person.css';

const Person = ({individual, removeStudent, studentCheck}) => {
  const button = studentCheck === 'student' ? <button onClick={() => removeStudent(individual.id)}>Remove</button> : '';
  const editableCheck = studentCheck === 'student' ? true : false;
  
  return (
    <section className="personContainer" key={individual.id}>
      <img className="image" src={individual.photo} alt={individual.name} />
      <h3 className="name" contentEditable={editableCheck} suppressContentEditableWarning={true}>{individual.name}</h3>
      <p className="quote" contentEditable={editableCheck} suppressContentEditableWarning={true}>{individual.quote}</p>
      <p className="superlative" contentEditable={editableCheck} suppressContentEditableWarning={true}>{individual.superlative}</p>
      {button}
    </section>
  )
}

export default Person;
