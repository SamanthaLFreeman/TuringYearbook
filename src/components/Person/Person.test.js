import React from 'react';
import { shallow } from 'enzyme';
import Person from './Person';

describe('Person', () => {
  it('should match the snapshot with all data passed in correctly if they are a student', () => {
    const person = { 
      id: 1, 
      name: 'Sam', 
      quote: 'Huh?', superlative: 'Most likely to', 
      studentCheck: 'student'
    }
    const wrapper = shallow(<Person individual={person} removeStudent={jest.fn()} studentCheck={'student'}/>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with all data passed in correctly if they are staff', () => {
    const person = {
      id: 1,
      name: 'Robbie',
      quote: 'pants...', superlative: 'Most likely to',
      studentCheck: 'staff'
    }
    const wrapper = shallow(<Person individual={person} removeStudent={jest.fn()} studentCheck={'staff'} />)

    expect(wrapper).toMatchSnapshot()
  })

})