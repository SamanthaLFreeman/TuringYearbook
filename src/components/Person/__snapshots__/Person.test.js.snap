// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot with all data passed in correctly if they are a student 1`] = `
<section
  className="personContainer"
  key="1"
>
  <img
    alt="Sam"
    className="image"
  />
  <h3
    className="name"
    contentEditable={true}
    suppressContentEditableWarning={true}
  >
    Sam
  </h3>
  <p
    className="quote"
    contentEditable={true}
    suppressContentEditableWarning={true}
  >
    Huh?
  </p>
  <p
    className="superlative"
    contentEditable={true}
    suppressContentEditableWarning={true}
  >
    Most likely to
  </p>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</section>
`;

exports[`Person should match the snapshot with all data passed in correctly if they are staff 1`] = `
<section
  className="personContainer"
  key="1"
>
  <img
    alt="Robbie"
    className="image"
  />
  <h3
    className="name"
    contentEditable={false}
    suppressContentEditableWarning={true}
  >
    Robbie
  </h3>
  <p
    className="quote"
    contentEditable={false}
    suppressContentEditableWarning={true}
  >
    pants...
  </p>
  <p
    className="superlative"
    contentEditable={false}
    suppressContentEditableWarning={true}
  >
    Most likely to
  </p>
</section>
`;
